[CI] Added gfx942 architecture to the 'Tune MLIR kernels' stage

stefankoncarevic opened the pull request:
for weekly CI and removed gfx1030 due to its decreased relevance.

stefankoncarevic committed:
[CI] Added gfx942 architecture to the 'Tune MLIR kernels' stage
for weekly CI and removed gfx1030 due to its decreased relevance.

umangyadav commented on mlir/utils/jenkins/Jenkinsfile:
```diff
@@ -723,7 +723,7 @@ pipeline {
                     axis {
                         name 'ARCH'
-                        values 'gfx908', 'gfx90a', 'gfx1030'
+                        values 'gfx908', 'gfx90a', 'gfx942'
```
Can you kick weekly run on this branch and see if gfx942 is being picked and it runs successfully ?

stefankoncarevic committed:
Add GFX942 to PerfDB archiving.

stefankoncarevic committed:
Fix formatting.

stefankoncarevic commented:
**Tuning time observed on the mi300 machine.**

**Link to Jenkins CI**: http://ml-ci-internal.amd.com:8080/blue/organizations/jenkins/MLIR%2Fmlir/detail/PR-1733/10/pipeline/387/

**Tuning time on mi300**:
Exhaustive tuning times on the mi300 machine are significantly longer compared to the mi200 machine:
**gemm**: 7 hours (mi300) vs. 1.5 hours (mi200)
**conv**: 18 hours (mi300) vs. 3.5 hours (mi200)
**attention**: 1.5 hours (mi300) vs. 20 minutes (mi200)
This represents a roughly 5x slowdown on the mi300 machine.

Investigation revealed that the machine sh5-1w300-h1-4.amd.com only has one GPU card when run rocm-smi, mi200 have 8 GPU.
This single GPU is likely the primary reason for the observed time slowdown.

umangyadav commented:
We should also add `gfx942` to test and benchmarking stage for the nightly runs.

umangyadav commented:
https://github.com/ROCm/rocMLIR-internal/issues/1738 

Associated issue

stefankoncarevic commented:
Yes, I just ask you. I will add also that part.

stefankoncarevic committed:
Added MI300 for benchmark and performance stage in nightly runs.

umangyadav commented on mlir/utils/jenkins/Jenkinsfile:
```diff
@@ -812,7 +813,7 @@ pipeline {
                     axis {
                         name 'CHIP'
-                        values 'gfx908', 'gfx90a', 'gfx1030', 'gfx1101'
+                        values 'gfx908', 'gfx90a', 'gfx942', 'gfx1030', 'gfx1101'
```
Can you kick nightly run on this pull request with only MI300 enabled and see if performance reports are being generated correctly or not ?

stefankoncarevic committed:
Updated reportUtils.py to replace deprecated styler.applymap
with styler.map.
Modified perfRunner.py to replace 'NaN' with np.nan in fillna
calls to avoid Warnings.

stefankoncarevic committed:
Updated archivePerfDB to handle only gfx908, gfx90a, and gfx942
